src/core/tests/type_prop/multinomial.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/multinomial.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/type_prop.hpp"
+
+using namespace testing;
+
+class TypePropMultinomialV13Test : public TypePropOpTest<ov::op::v13::Multinomial> {};
+
+TEST_F(TypePropMultinomialV13Test, input_probs_f64_num_samples_i32_convert_i32) {
+    const auto probs = std::make_shared<ov::op::v0::Parameter>(ov::element::f64, ov::Shape{4});
+    const auto num_samples = std::make_shared<ov::op::v0::Parameter>(ov::element::i32, ov::Shape{1});
+    const auto op = make_op(probs, num_samples, ov::element::i32, false, false, 0, 0);
+    EXPECT_EQ(op->get_element_type(), ov::element::i32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (ov::PartialShape::dynamic(1)));
+}
+
+TEST_F(TypePropMultinomialV13Test, input_probs_f32_num_samples_i32_convert_i64) {
+    const auto probs = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, ov::Shape{4, 4});
+    const auto num_samples = std::make_shared<ov::op::v0::Parameter>(ov::element::i32, ov::Shape{});
```

Comment:
Would be good to add here a test with `num_samples` as a Constant (not only Parameter).
Can be done separately.

Reply:
Added in CPU PR: https://github.com/openvinotoolkit/openvino/pull/20406